connector/src/main/scala/com/microsoft/kusto/spark/datasink/SparkIngestionProperties.scala

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.kusto.spark.datasink
+
+import java.util
+
+import com.microsoft.azure.kusto.ingest.IngestionProperties
+import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility
+import org.codehaus.jackson.annotate.JsonMethod
+import org.codehaus.jackson.map.ObjectMapper
+
+class SparkIngestionProperties {
+  var flushImmediately: Boolean = false
+  var dropByTags: util.ArrayList[String] = _
+  var ingestByTags: util.ArrayList[String] = _
+  var additionalTags: util.ArrayList[String] = _
+  var ingestIfNotExists: util.ArrayList[String] = _
+  var creationTime: String = _ //TODO: should be date? what are the ok formats?
+  var csvMapping: String = _
+  var csvMappingNameReference: String = _
+
+  override def toString: String = {
+    new ObjectMapper().setVisibility(JsonMethod.FIELD, Visibility.ANY)
+      .writerWithDefaultPrettyPrinter
+      .writeValueAsString(this)
+  }
+}
+
+object SparkIngestionProperties {
+  private[kusto] def fromStringToIngestionProperties(json: String, database: String, table: String) = {
```

Comment:
should probably be part of java client instead of spark.

also, naming wise, should be `fromString`. no need to mention the to because of the context (static class).

Reply:
the fromStringToIngestionProperties 
means that from a JSON string of SparkIngestionProperties  i create IngestionProperties

Reply:
how about `fromString(json,database, table).toIngestionProperties()`?

also, I'm sure why you need to go trough `SparkIngestionProperties`, shouldn't it be possible to load `IngestionProperties` from json?